index.js

```diff
@@ -1,15 +1,25 @@
 'use strict';
 
-var jsRegex = /^(%20)*javascript:.*/im;
-var dataRegex = /^(%20)*data:.*/im;
+var invalidPrototcolRegex = /^(%20)*(javascript|data)/im;
 var ctrlCharactersRegex = /[^\x20-\x7E]/gmi;
+var urlSchemeRegex = /^(.+):/gm;
```

Comment:
Pretty sure this regex is wrong.

```js
> 'http://example.com:3000'.match(urlSchemeRegex)
// [ 'http://example.com:' ]

> 'http://example.com/path/to:something'.match(urlSchemeRegex)
// [ 'http://example.com/path/to:' ]

> 'http://example.com/path/to:something:else'.match(urlSchemeRegex)
// [ 'http://example.com/path/to:something:' ]
```

Maybe this is desired, though.

I think you want the regex to be something like `/^([^:]+):/gm` or parse the protocol with a more robust method (like [using an `a` tag](https://evanhahn.com/parse-urls-with-a-tags/) or a [URL module](https://nodejs.org/api/url.html)).

Reply:
Good call on the a tag, we'll do that.

Reply:
Actually, we should be able to use this in a node context as well. So we can't use the `a` tag method.